libnavigation-base/src/main/java/com/mapbox/navigation/base/route/model/RouteProgressStateNavigation.kt

```diff
@@ -0,0 +1,9 @@
+package com.mapbox.navigation.base.route.model
+
+enum class RouteProgressStateNavigation {
```

Comment:
What about using [`sealed class`es](https://kotlinlang.org/docs/reference/sealed-classes.html#sealed-classes) instead of `enum`s?

Reply:
`Sealed` class makes sense when your `enums` need to represent some `data`. In our case they are just simple enums. I would rather have it as it is. [Reference](https://proandroiddev.com/kotlin-sealed-classes-enums-with-swag-d3c4b799bcd4)